### ISSUE TITLE
Why dispersion loss uses val/test data to initialize prototypes

### ISSUE THREAD

**Issue by zjysteven** [deeplearning-wisc/cider#9]
Hi,

I was wondering why `val_loader` (which essentially loads test data) is used in Dispersion Loss module to initialize prototypes.
https://github.com/deeplearning-wisc/cider/blob/3d234f6cc2ca28a150b7b8288ff5009fa60949d1/train_cider.py#L164

This move would leak information about test data, although may not to a great extent since it's just initialization.

**Comment by alvinmingwisc** [comment]
Thanks for the point! We used the val loader simply because initialization is faster. Prototype initialization does not matter much in practice, especially when the model itself is also randomly initialized at the beginning. To be more rigorous, one can use the training set or a subset of the training set to initialize the class prototypes.

**Comment by zjysteven** [comment]
Thanks that's what I was thinking too. I'd also appreciate some clarification on the training setup for ImageNet-100. In the paper it seems to be fine-tuning for 10 epochs with 0.01 as the learning rate, while in the script it seems to be fine-tuning for 20 epochs with 0.5 as the learning rate.

**Comment by alvinmingwisc** [comment]
Yes, we observed that fine-tuning 10 epochs with 0.01 init lr gives better performance, as in the paper. Fine-tuning longer does not make much difference. 0.5 is the default lr when the model is trained from scratch, but might be too high in the fine-tuning scheme. Will double-check and update the script to specify init lr for the fine-tuning setting.

**Comment by zjysteven** [comment]
Cool thanks!

**Comment by alvinmingwisc** [comment]
Empirically, using the ID training set to initialize class prototypes does not seem to have much impact with various hyperparameter configurations & OOD test scores. Here are some new results I get: 
 
<img width="509" alt="Screen Shot 2023-04-27 at 3 21 32 PM" src="https://user-images.githubusercontent.com/20054386/234982006-595f7a15-70f1-4a10-b9bd-3523c8b3b7c6.png">
<img width="507" alt="Screen Shot 2023-04-27 at 3 21 37 PM" src="https://user-images.githubusercontent.com/20054386/234982078-de8f29d2-cf7e-42d4-8d03-5ec88c0486bb.png">

**Comment by zjysteven** [comment]
Looks great! Thanks for the additional results.
